opts/env.go

```diff
@@ -29,6 +30,28 @@ func ValidateEnv(val string) (string, error) {
 	return fmt.Sprintf("%s=%s", val, os.Getenv(val)), nil
 }
 
+// ExpandEnvWildcard takes a glob pattern and then returns a slice of strings
+// of the form KEY=VALUE with environment variables where KEY matches the glob
+// pattern.
+func ExpandEnvWildcard(pattern string) []string {
+	var opts []string
+
+	for _, entry := range os.Environ() {
+		parts := strings.SplitN(entry, "=", 2)
+		name := parts[0]
+		val := parts[1]
```

Comment:
In actual implementation `entry` contains `=`, but I thought the godoc is not clear enough.

Maybe just adding a comment like `if value is empty, s is like "K=", not "K".` is enough

https://github.com/docker/docker/pull/30781#discussion_r99892679